### DISCUSSION TITLE
Is the sahi input form array for yolov5 forward?

### DISCUSSION THREAD

**Discussion by YoungjaeDev** [obss/sahi#659] [Q&A] [answered]
Hello
Can you tell me why it's not Torch but numpy?
Is it possible if it's np input?

**Comment by YoungjaeDev** [comment]
Ah because of `AutoShape` default True

What is the difference between AutoShape and basic yolov5 modules?

Thank you!

**Comment by fcakyon** [comment, marked as answer]
We wanted `sahi` to work independent from torch thats why all detectors accept numpy or str as image input 👍

**Comment by fcakyon** [comment]
> What is the difference between AutoShape and basic yolov5 modules?

You should ask the yolov5 related questions in ultralytics/yolov5 repo 🚀
